include/grpcpp/ext/otel_plugin.h

```diff
@@ -142,6 +152,10 @@ class OpenTelemetryPluginBuilder {
   /// Registers a global plugin that acts on all channels and servers running on
   /// the process.
   absl::Status BuildAndRegisterGlobal();
+  /// Builds an open telemetry plugin, returns the plugin object when succeeded
+  /// or an error status when failed.
+  GRPC_MUST_USE_RESULT absl::StatusOr<std::shared_ptr<OpenTelemetryPlugin>>
```

Comment:
any special reason why this returns a shared_ptr?

Reply:
In the approach I was imagining, the returned object would have the `shared_ptr` member.